x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/jinaai/JinaAIServiceSettings.java

```diff
@@ -0,0 +1,180 @@
+
+public class JinaAIServiceSettings extends FilteredXContentObject implements ServiceSettings, JinaAIRateLimitServiceSettings {
+
+    public static final String NAME = "jinaai_service_settings";
+    public static final String OLD_MODEL_ID_FIELD = "model";
+    public static final String MODEL_ID = "model_id";
+    private static final Logger logger = LogManager.getLogger(JinaAIServiceSettings.class);
+    public static final RateLimitSettings DEFAULT_RATE_LIMIT_SETTINGS = new RateLimitSettings(2_000);
+
+    public static JinaAIServiceSettings fromMap(Map<String, Object> map, ConfigurationParseContext context) {
+        ValidationException validationException = new ValidationException();
+
+        String url = extractOptionalString(map, URL, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        URI uri = convertToUri(url, URL, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        String oldModelId = extractOptionalString(map, OLD_MODEL_ID_FIELD, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        RateLimitSettings rateLimitSettings = RateLimitSettings.of(
+            map,
+            DEFAULT_RATE_LIMIT_SETTINGS,
+            validationException,
+            JinaAIService.NAME,
+            context
+        );
+
+        String modelId = extractOptionalString(map, MODEL_ID, ModelConfigurations.SERVICE_SETTINGS, validationException);
+
+        if (context == ConfigurationParseContext.REQUEST && oldModelId != null) {
+            logger.info("The JinaAI [service_settings.model] field is deprecated. Please use [service_settings.model_id] instead.");
+        }
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new JinaAIServiceSettings(uri, modelId(oldModelId, modelId), rateLimitSettings);
+    }
+
+    private static String modelId(@Nullable String model, @Nullable String modelId) {
+        return modelId != null ? modelId : model;
+    }
+
+    private final URI uri;
+    private final String modelId;
+    private final RateLimitSettings rateLimitSettings;
+
+    public JinaAIServiceSettings(@Nullable URI uri, @Nullable String modelId, @Nullable RateLimitSettings rateLimitSettings) {
+        this.uri = uri;
+        this.modelId = modelId;
+        this.rateLimitSettings = Objects.requireNonNullElse(rateLimitSettings, DEFAULT_RATE_LIMIT_SETTINGS);
+    }
+
+    public JinaAIServiceSettings(@Nullable String url, @Nullable String modelId, @Nullable RateLimitSettings rateLimitSettings) {
+        this(createOptionalUri(url), modelId, rateLimitSettings);
+    }
+
+    public JinaAIServiceSettings(StreamInput in) throws IOException {
+        uri = createOptionalUri(in.readOptionalString());
+        modelId = in.readOptionalString();
+
+        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_15_0)) {
```

Comment:
This transport version check can be removed as soon as you introduced the new transport version mentioned above. This is just needed for providers, which were introduced before rate limiting.